src/GafferScene/ShaderTweaks.cpp

```diff
@@ -316,8 +323,66 @@ bool ShaderTweaks::applyTweaks( IECoreScene::ShaderNetwork *shaderNetwork, Tweak
 				{
 					throw IECore::Exception( fmt::format( "Cannot apply tweak to \"{}\" : Mode must be \"Replace\" when inserting a connection", name ) );
 				}
+
 				const auto inputParameter = ShaderNetworkAlgo::addShaders( shaderNetwork, inputNetwork );
 				shaderNetwork->addConnection( { inputParameter, parameter } );
+
+				// TODO - it would be substantially more efficient to search for and process tweak sources
+				// just in `inputNetwork` before merging it to `shaderNetwork` ... but this would require
+				// dealing with weird connections where the input node handle is relative to `shaderNetwork`,
+				// but the output handle is relative to `inputNetwork`. This gets very confusing if there are
+				// nodes in the two networks with the same name, which get uniquified during addShaders.
+				// Doing this after merging simplifies all that.
+				std::vector<IECore::InternedString> shadersToDelete;
+				for( const auto &i : shaderNetwork->shaders() )
+				{
+					if( i.second->getName() == ShaderTweakProxy::shaderTweakProxyIdentifier() )
+					{
+						shadersToDelete.push_back( i.first );
+
+						ShaderNetwork::ConnectionRange range = shaderNetwork->outputConnections( i.first );
+						const std::vector<ShaderNetwork::Connection> outputConnections( range.begin(), range.end() );
+
+						for( const auto &c : outputConnections )
+						{
+							ShaderNetwork::Parameter dest = c.destination;
+
+							StringData *sourceNodeData = nullptr;
+							try
+							{
+								sourceNodeData = IECore::runTimeCast< StringData >(
+									i.second->parameters().at( "shaderTweakProxySourceNode" ).get()
+								);
+							}
+							catch( ... )
+							{
+							}
```

Comment:
This could be a one-liner without any need for exception suppression :

```const auto sourceNodeData = i.second->parametersData()->member<StringData>( "shaderTweakProxySourceNode" )```

Reply:
Addressed